src/com/vivian/java/leetcode/GraphNode.java

```diff
@@ -0,0 +1,23 @@
+package com.vivian.java.leetcode;
+import com.vivian.java.collections.ArrayList;
+
+public class GraphNode<T> {
+    private T data;
+    private ArrayList<GraphNode<T>> neighbors = new ArrayList<>();
```

Comment:
Use List<GraphNode<T>> as the reference type, don't use concrete ArrayList. 
This is to program to an interface, instead of implementation. The reason is that when you want to change the actual implementation type to something else, for example, LinkedList, you don't need to change the reference type and the code below this line.

Reply:
solved